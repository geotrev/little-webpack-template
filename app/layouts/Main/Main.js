import React from "react"
import { Route, Switch } from "react-router-dom"

import { rootPath, aboutPath } from "app/routes"
import Nav from "app/components/Nav"
import Home from "app/pages/Home"
import About from "app/pages/About"
import "./styles.scss"

const renderAbout = () => <About message="Woohoo, you did it." />

export default function Main() {
  return (
    <div id="site">
      <header>
        <Nav />
      </header>
      <main id="main">
        <Switch>
          <Route path={aboutPath} render={renderAbout} />
          <Route path={rootPath} component={Home} />
        </Switch>
      </main>
    </div>
  )
}
