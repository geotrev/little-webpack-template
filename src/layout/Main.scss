body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

.emoji {
  display: block;
  margin: 0 auto;
  font-size: 4rem;
}

#home, #about {
  flex: 1 1 auto;
  max-width: 100%;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  font-size: 16px;
  text-align: center;
  padding: 120px 20px;

  .hopping {
    animation: hop 1s alternate infinite;
  }
  
  .waving {
    animation: wave 0.75s alternate infinite;
  }
  
  h1 {
    font-weight: bold;
    font-size: 3rem;
  }

  p {
    font-size: 1.5rem;
  }
}

@keyframes wave {
  100% {
    transform: rotate(20deg);
  }
}

@keyframes hop {
  0% {
    margin: -15px 0 15px 0;
  }

  80%, 100% {
    margin: 0;
  }

  10% {
    margin: -15px 0 15px 0;
  }
}
