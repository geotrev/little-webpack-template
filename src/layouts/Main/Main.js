import React from "react"
import { Route, Switch } from "react-router-dom"

import { rootPath, aboutPath } from "routes"
import Nav from "components/Nav"
import Home from "pages/Home"
import About from "pages/About"
import "./styles.scss"

const renderAbout = () => <About message="Woohoo, you did it." />

export default function Main() {
  return (
    <div id="site">
      <header>
        <Nav />
      </header>
      <main>
        <Switch>
          <Route path={aboutPath} render={renderAbout} />
          <Route path={rootPath} component={Home} />
        </Switch>
      </main>
    </div>
  )
}
