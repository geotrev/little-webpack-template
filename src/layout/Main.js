import React from 'react';
import './Main.scss';
import { Route, Switch } from 'react-router-dom';
import { Nav } from 'components';
import { Home, About } from 'pages';
/*
** ---------------------------------------------------
**
** --> PWA Icon Generator:
**     https://app-manifest.firebaseapp.com/
**
** --> Favicon Generator:
**     http://www.favicomatic.com/
**
** ---------------------------------------------------
*/
import 'assets/icons/favicon.ico';
import 'assets/icons/favicon-32x32.png';

const Main = () => (
  <div id="site">
    <header>
      <Nav />
    </header>
    <main>
      <Switch>
        <Route path='/about' component={ About } />
        <Route path='/' component={ Home } />
      </Switch>
    </main>
  </div>
)

export default Main
