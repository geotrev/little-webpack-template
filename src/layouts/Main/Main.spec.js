import React from "react"
import Main from "./Main"
import { BrowserRouter as Router } from "react-router-dom"

jest.mock("components/Nav", () => global.simpleMock("Nav"))
jest.mock("pages/Home", () => global.simpleMock("Home"))
jest.mock("pages/About", () => global.simpleMock("About"))

describe("<Main />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(
      <Router>
        <Main />
      </Router>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
