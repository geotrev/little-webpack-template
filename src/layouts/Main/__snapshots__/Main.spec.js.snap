// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> matches snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Main>
      <div
        id="site"
      >
        <header>
          <Nav />
        </header>
        <main>
          <Switch>
            <Route
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            >
              <Home
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              />
            </Route>
          </Switch>
        </main>
      </div>
    </Main>
  </Router>
</BrowserRouter>
`;
